packages/cli-server-api/src/__tests__/securityHeaderesMiddleware.test.ts

```diff
@@ -0,0 +1,54 @@
+import securityHeadersMiddleware from '../securityHeadersMiddleware';
+
+describe('securityHeadersMiddleware', () => {
+  let req, res, next;
+
+  beforeEach(() => {
+    req = {
+      headers: {},
+    };
+    res = {
+      setHeader: () => {},
+    };
+    next = jest.fn();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('it should block requests from different origins', () => {
+    req.headers.origin = 'https://example.com';
+    const middleware = securityHeadersMiddleware({});
+    middleware(req, res, next);
+    expect(next).toHaveBeenCalledWith(expect.any(Error));
+  });
+
+  it('it should allow requests from localhost', () => {
+    req.headers.origin = 'http://localhost:3000';
+    const middleware = securityHeadersMiddleware({});
+    middleware(req, res, next);
+    expect(next).toHaveBeenCalled();
+  });
+
+  it('it should allow requests from devtools', () => {
+    req.headers.origin = 'devtools://devtools';
+    const middleware = securityHeadersMiddleware({});
+    middleware(req, res, next);
+    expect(next).toHaveBeenCalled();
+  });
+
+  it('it should allow requests from custom host if provided in options', () => {
+    req.headers.origin = 'http://customhost.com';
+    const middleware = securityHeadersMiddleware({host: 'customhost.com'});
+    middleware(req, res, next);
+    expect(next).toHaveBeenCalled();
+  });
+
+  it('it should block requests from custom host if provided in options but not matching', () => {
```

Comment:
very minor nit: all test titles start with "it" which is kinda redundant 😅

Reply:
Oh good shout. Will fix that up!